protos/feast/core/JobService.proto

```diff
@@ -0,0 +1,203 @@
+
+option go_package = "github.com/feast-dev/feast/sdk/go/protos/feast/core";
+option java_outer_classname = "JobServiceProto";
+option java_package = "feast.proto.core";
+
+import "google/protobuf/timestamp.proto";
+import "feast/core/DataSource.proto";
+import "feast/serving/ServingService.proto";
+
+service JobService {
+    // Start job to ingest data from offline store into online store
+    rpc StartOfflineToOnlineIngestionJob (StartOfflineToOnlineIngestionJobRequest) returns (StartOfflineToOnlineIngestionJobResponse);
+
+    // Produce a training dataset, return a job id that will provide a file reference
+    rpc GetHistoricalFeatures (GetHistoricalFeaturesRequest) returns (GetHistoricalFeaturesResponse);
+
+    //   IMPORTANT: The following RPCs are not must-haves for the MVP. They are stretch goals. All of these methods would exist internally in the job service and would be called by a control loop to start/stop/restart streaming ingestion jobs.
+
+    // Start job to ingest data from stream into online store
+    rpc StartStreamToOnlineIngestionJob (StartStreamToOnlineIngestionJobRequest) returns (StartStreamToOnlineIngestionJobResponse);
+
+    // List all types of jobs
+    rpc ListJobs (ListJobsRequest) returns (ListJobsResponse);
+
+    // Stop a single job
+    rpc StopJob (StopJobRequest) returns (StopJobResponse);
+
+    // Get details of a single job
+    rpc GetJob (GetJobRequest) returns (GetJobResponse);
+}
+
+
+enum JobType {
+	INVALID_JOB = 0;
+	OFFLINE_TO_ONLINE_JOB = 1;
+	STREAM_TO_ONLINE_JOB = 2;
+	EXPORT_JOB = 4;
+}
+
+enum JobStatus {
+	JOB_STATUS_INVALID = 0;
+       // The Job has be registered and waiting to get scheduled to run
+	JOB_STATUS_PENDING = 1;
+       // The Job is currently processing its task
+	JOB_STATUS_RUNNING = 2;
+       // The Job has successfully completed its task
+	JOB_STATUS_DONE = 3;
+       // The Job has encountered an error while processing its task
+	JOB_STATUS_ERROR = 4;
+}
+
+message Job {
+  // Identifier of the Job
+  string id = 1;
+  // External Identifier of the Job assigned by the Spark executor
+  string external_id = 2;
+  // Type of the Job
+  JobType type = 3;
+  // Current job status
+  JobStatus status = 4;
+  // Timestamp on when the job was is created
+  google.protobuf.Timestamp created_time = 5;
+  // Timestamp on when the job has stopped.
+  google.protobuf.Timestamp stop_time = 6;
+
+  message ExportJobMeta {
+    // Glob of the exported files that should be retrieved to reconstruct
+    // the dataframe with retrieved features.
+    repeated string file_glob = 1;
+    // The Historical Features request that triggered this export job
+    GetHistoricalFeaturesRequest request = 2;
+  }
+
+  message OfflineToOnlineMeta {
+    // Reference to the Feature Table being populated by this job
+    string project = 1;
+    string table_name = 2;
+    // Connection String to the Redis Online Store that the job is writing to.
+    string redis_connection_string = 3;
```

Comment:
we should make this a struct for consistency with the[ ingestion job parameter format](https://github.com/feast-dev/feast/blob/master/spark/ingestion/src/main/scala/feast/ingestion/IngestionJobConfig.scala#L29)

Reply:
Actually lets just drop it from here for now. The only way this entire `Job` proto is used is to list jobs, i think it is fine if you just get the basic info like job type and table name there, but not redis connection details. 

We can always add more stuff later.